### DISCUSSION TITLE
Alerts - Reduced plot counts and reduced connected workers

### DISCUSSION THREAD

**Discussion by SquareMesh** [guydavis/machinaris#927] [General]
Looking for alerts that will 
- notify me if a worker has gone offine
- notify me if my plot count is reduced below its highest watermark (per worker)

In my use of Machinaris I have 13 machines (11 use machinaris, other 2 are windows), one machine is the main set to farming and the rest are harvesters.  (I know too many machines, but it is my chosen path for this hobby / adventure)

Issue #1 - Occassionally a USB drive or branch of USB drives go wonky and disappear from the USB bus and their plots disappear and reduce the overall plot count.  Without an alert the problem continues unattended until I happen to check the plot counts and discover mismatch.

Issue #2 - Reboot of server.  Windows is impacted by the periodic update cycle as reboot doesn't restart Chia harvester.  Also Linux Server (Unraid) can reboot on occassional basis and does not remount disks/auto start Docker. Without an alert the rebooted server remains unattended.

Workaround that I currently employ is to regularly check Machinaris Summary for a change in workers or a change in plot counts.

**Comment by guydavis** [comment]
Good day,

For your first concern, the Chiadog project [should alert on decreasing plot count](https://github.com/martomi/chiadog/blob/main/src/chia_log/handlers/condition_checkers/non_decreasing_plots.py#L12).  Many users have experienced sporadic results from this third-party tool however.

For your second issue, the `docker-compose.yml` generated at https://www.machinaris.app/, automatically includes `restart: always`, so if Docker is correctly configured, then all containers will be restarted on reboot.  This applies to both Windows and Linux.

Thanks for the feedback.

**Reply by SquareMesh** [reply]
Hi there,

Appreciate the reply.  Will do some research on Chiadog to see how this can be used.  

On the 2nd issue the server reboot in Unraid is less an issue with Docker it is more an issue with Unraid and on reboot it does not auto-mount disks and start up docker.  Once disks are manually started then Docker auto-starts and starts the Machinaris container fine. Having alert of missing worker is a nice to have really, thought it could be implemented based on Summary already knowing [Workers Active] / [Workers Total Known].
